TLS Ladder Fixes (Dead Center to No Mercy)

Derpduck opened the pull request:
## Ladder Fixes

Fixes various issues with ladders added in The Last Stand update including:
* De-synced / "laggy" movement when climbing, issue increases with ping
* Ladders lower than the object they are used to climbed on
* Ladders extending out too far past a corner
* Getting pushed off or stuck due to geometry
* Re-oriented ladders that were placed adjacent to the object they climb on to match typical Valve ladders 
* Added clipping to prevent getting stuck on some problematic ladders

### c1m2_streets (#197)
* `_ladder_postdropcut_cloned_turnpikemid` - Fixed movement issues
* `_ladder_tentbus_cloned_stairsbus` - Fixed movement issues
	* Renamed to: `_ladder_tentbus_cloned_startfenceback`
* `_ladder_turnpikeright_cloned_turnpikemid` - Fixed movement issues

### c2m1_highway
* `_ladder_barrelsemi_cloned_caralarmshort` - Fixed movement issues
* `_ladder_endbusback_cloned_caralarmshort` - Fixed movement issues
	* Renamed to: `_ladder_endbusback_cloned_caralarmbus`
* `_ladder_endbusfront_cloned_caralarmshort` - Fixed movement issues
	* Renamed to: `_ladder_endbusfront_cloned_caralarmbus`
* `_ladder_qolbus_cloned_caralarmshort` - Fixed movement issues
	* Renamed to: `_ladder_qolbus_cloned_caralarmbuspool`
* `_ladder_shrubberybus_cloned_caralarmshort` - Fixed movement issues
	* Renamed to: `_ladder_shrubberybus_cloned_caralarmbus`
* `_ladder_slopetruck_cloned_caralarmshort` - Fixed movement issues
* `_ladder_tankfightsemi_cloned_caralarmshort` - Fixed movement issues
* `_ladder_motelroofright_cloned_onewaycliff` + `_ladder_motelroofright_pipe` - Removed redundant duplicated ladders / props

### c2m5_concert
* `_ladder_startfenceback_cloned_fencecoverfront` - Fixed movement issues
* `_ladder_startfencefront_cloned_fencecoverback` - Fixed movement issues

### c3m1_plankcountry
* `_ladder_startsemitire_cloned_boardwalkhole` - Fixed movement issues
* `_ladder_starttrainfront1_cloned_boardwalkhole` - Fixed movement issues
	* Renamed to: `_ladder_starttrainfront1_cloned_boatshack`
* `_ladder_starttrainfront2_cloned_boardwalkhole` - Fixed movement issues

### c3m2_swamp
* `_ladder_endbarricadeleft1_cloned_airplaneleft` + `_ladder_endbarricadeleft2_cloned_airplaneleft` - Fixed movement issues
* `_ladder_endbarricaderight1_cloned_airplaneleft` + `_ladder_endbarricaderight2_cloned_airplaneleft` - Fixed movement issues

### c3m3_shantytown
* `_ladder_shortcuttrailerfront_cloned_shortcuttrailerback` - Fixed movement issues
	* Renamed to: `_ladder_shortcuttrailerfront_cloned_privateproperty`
* `_ladder_startsafehouse_cloned_highgenerator` - Fixed movement issues
	* Renamed to: `_ladder_startsafehouse_cloned_shantyshop`

### c4m1_milltown_a
* Added `_ladder_cornerhome_clip` - Prevents getting stuck on the roof when climbing up `_ladder_cornerhomeplants_cloned_garagesalehome`
* `_ladder_grocerypillar_cloned_garagesalehouse` - Fixed ladder being too low down
* `_ladder_tallbuildingright_cloned_tallbuildingleft` - Fixed ladder being too low down

### c4m2_sugarmill_a
* `_ladder_parkoursiloleft_cloned_marshtrailer` + `_ladder_parkoursiloright_cloned_marshtrailer` - Fixed movement issues
	* Renamed to: `_ladder_parkoursiloleft_cloned_stacksfencebackr` & `_ladder_parkoursiloright_cloned_stacksfenceback` respectively
* `_ladder_silosemirearleft_cloned_millgrinder` - Fixed movement issues
* `_ladder_silosemirearright_cloned_millgrinder` - Fixed movement issues
* `_ladder_ducatelroofl_cloned_brokenlocker` + `_ladder_ducatelroofr_cloned_brokenlocker` - Fixed movement issues

### c4m3_sugarmill_b
* `_ladder_parkoursiloleft_cloned_marshtrailer` + `_ladder_parkoursiloright_cloned_marshtrailer` - Fixed movement issues
	* Renamed to: `_ladder_parkoursiloleft_cloned_stacksfencebackr` & `_ladder_parkoursiloright_cloned_stacksfenceback` respectively
* `_ladder_silosemirearleft_cloned_millgrinder` - Fixed movement issues
* `_ladder_silosemirearright_cloned_millgrinder` - Fixed movement issues
* `_ladder_ducatelroofl_cloned_brokenlocker` + `_ladder_ducatelroofr_cloned_brokenlocker` - Fixed movement issues

### c4m4_milltown_b
* Added `_ladder_cornerhome_clip` - Prevents getting stuck on the roof when climbing up `_ladder_cornerhomeplants_cloned_garagesalehome`
* `_ladder_tallbuildingright_cloned_tallbuildingleft` - Fixed ladder being too low down

### c4m5_milltown_escape
* `_ladder_grocerypillar_cloned_garagesalehouse` - Fixed ladder being too low down

### c5m1_waterfront
* `_ladder_endbluehouse_cloned_waterfrontbrick` - Re-oriented ladder against building instead of wall
	* Adjusted clip `_ladder_endbluehouse_clipleft` accordingly
		* Renamed to: `_ladder_endbluehouse_clip`
	* Removed now redundant clip: `_ladder_endbluehouse_clipwall`
	* Added prop `_ladder_endgutterbluehouse_pipe` for ladder

### c5m2_park
* `_ladder_billboardleft_cloned_watchtower` + `_ladder_billboardright_cloned_watchtower` - Fixed movement issues
	* Replaced clip: `_ladder_billboard_clip` with: `_ladder_billboard_clip_a` + `_ladder_billboard_clip_b`
* `_ladder_hmmwvawning_cloned_startcopcarfence` - Fixed movement issues
	* Renamed to: `_ladder_hmmwvawning_cloned_cedatrailerfence`
* `_ladder_overpasshigh_cloned_cedatrailerfence` - Fixed getting stuck on geometry / pushed off by geometry

### c5m3_cemetery
* `_ladder_1stdumpster1_cloned_stuckfencefront` - Fixed movement issues
	* Renamed to: `_ladder_1stdumpster1_cloned_stuckfenceback`
* `_ladder_1stdumpster2_cloned_stuckfencefront` - Fixed movement issues
	* Renamed to: `_ladder_1stdumpster2_cloned_stuckfenceback`
* `_ladder_fencecornerleft_cloned_fencebackeastr` - Re-oriented ladder against fence instead of wall
	* Renamed to: `_ladder_fencecornerleft_cloned_fencebacflamingo`
* `_ladder_fencecornerright_cloned_fencebacksouthl` - Re-oriented ladder against fence instead of wall
	* Renamed to: `_ladder_fencecornerright_cloned_fencebackheli`
* `_ladder_holefenceback_cloned_firstcrypt` - Fixed movement issues
* `_ladder_holefencefront_cloned_firstcrypt` - Fixed movement issues

### c5m4_quarter
* `_ladder_busjazzclub_cloned_startwhitefence` - Fixed movement issues
* `_ladder_endsemifront_cloned_endsemiback` - Moved onto the nearby wrecked semi trailer
	* Renamed to: `_ladder_endsemiwrecked_cloned_streetendfence`

### c5m5_bridge
* `_ladder_endcedatrailer_cloned_endchainlink` - Fixed movement issues
* `_ladder_endlosfence_cloned_backnodraw` - Fixed movement issues
* `_ladder_finalsidehouse_cloned_finalrungs` - Fixed movement issues

### c6m1_riverbank
* `_ladder_elecbox_cloned_bluebin` - Fixed getting pushed off the ladder by geometry
* `_ladder_upperbalconynear_cloned_upperbalconyfar` - Fixed ladder extending past the corner
	* Adjusted clip `_ladder_upperbalconynear_clip` accordingly

### c6m2_bedlam
* `_ladder_forkliftvines_cloned_vinefence` - Re-oriented ladder against fence instead of wall

### c7m2_barge
* `_ladder_comicboatsleftL_cloned_toolhouse` + `_ladder_comicboatsleftR_cloned_toolhouse` - Fixed movement issues
* `_ladder_comicwitchboat_cloned_toolhouse` - Fixed movement issues
* `_ladder_endbarricadeleft_cloned_vanishbarricade` - Fixed getting pushed off the ladder by geometry
* `_ladder_endbarricaderight_cloned_nomanssemi` - Fixed movement issues
* `_ladder_permstuckend_cloned_toolhouse` - Fixed movement issues
* `_ladder_tankpoleswallL_cloned_tankpolesfenceright` + `_ladder_tankpoleswallR_cloned_tankpolesfenceright` - Fixed movement issues

### c8m1_apartment
* `_ladder_flatnose_cloned_chaintilt` - Fixed movement issues

### c8m2_subway
* `_ladder_deadendrubble_cloned_endfenceshortest` - Fixed movement issues
* `_ladder_endquickroof_cloned_oneunitbroke` - Fixed movement issues
* `_ladder_innertanker_cloned_nodrawfence` - Fixed movement issues
* `_ladder_tanksubqol_cloned_copcarsbrick` - Fixed movement issues
* `_ladder_tanksubwreck_cloned_endfenceshortest` - Fixed movement issues
	* Removed clip: `_ladder_tanksubwreck_clip`
* `_ladder_tankwarprubble_cloned_warehousewindow` - Fixed movement issues

### c8m3_sewers
* `_ladder_overturnedsemiB_cloned_nodrawfence` - Fixed movement issues
* `_ladder_scissormini_cloned_sewerdropB` - Fixed ladder being too low down


## Clipping
### c1m3_mall
* `_commentary_shortcut_storeceiling` + `_commentary_shortcut_storevent` - Fixed commentary replacement clips in the Just For Kidz store blocking infected, now only blocks survivors
> Looks to be the only instance of incorrect team blocking from commentary blockers re-work

### c3m2_swamp
* Added `_permstuck_behindrocks` - Fixes a perma-stuck spot near the parachutist

### c3m3_shantytown
* Improve accuracy of clipping on saferoom roof

### c5m3_cemetery
* `_stuckwarp_staircase` - Fixed clip extending much higher than necessary


## Miscellaneous
* Replaced instances of "mapfixes" with "community_update" in `z_developer_showupdate.nut` and `functions.nut`

Tsuey commented:
Near perfection, so much that I really hate to nitpick -- but other than these, I'm ready to merge.

# Issues

### c2m1

- There's also a duplicate make_prop for `_ladder_motelroofright_pipe` to delete.

- These substitute ladders visibly hover about 30 units above ground (twice as much as fixes on **c3m1** start train and **c5m5** end CEDA trailer that aren't as obvious, so maybe let's draw the line at a max of 16-unit hover):

	`_ladder_barrelsemi_cloned_signfence`
	`_ladder_slopetruck_cloned_caralarmbus`
	`_ladder_tankfightsemi_cloned_signfence`

![image](https://user-images.githubusercontent.com/71397129/167391735-f733d8b4-70c5-4075-aee0-1a08009e9231.png)

### c3m2

- Rename `_permastuck_behindrocks` to `_permstuck_behindrocks` for consistency with the [other similar entities](https://github.com/Tsuey/L4D2-Community-Update/search?q=permstuck) (the Commentary Rework was perfect in this regard).

# Really not a problem

Technically **c1m3**'s `_ladder_tentbus_cloned_saferoomfence` would be best renamed to `_ladder_tentbus_cloned_startfenceback` to match the others that use the same source ladder.

Also **c2m1** `caralarmshort / signfence / caralarmbus` names could use a consistency check -- really though, these don't matter for older ladders, TLS dev was hectic so I'm sure there's plenty that are inconsistent (e.g. same source ladder is referred to by two different names), and fixing these is lowest of priority.

# Sidenotes for any readers

c5m4:

![image](https://user-images.githubusercontent.com/71397129/167392088-6528a49a-8ddb-4a62-af71-ba326f8c204a.png)

This ladder near the end saferoom:

`make_ladder( "_ladder_endsemifront_cloned_endsemiback", "1122.36 -2274.4004 146.7381", "2144 -4483 0", "0 180 0", "-0.96 0.28 0" );`

Was moved to a more useful spot nearby at:

`make_ladder( "_ladder_endsemiwrecked_cloned_streetendfence", "-510 -800 140.24", "500.15 -2789.49 -52", "0 168.5 0", "-0.97 0.19 0" );`

Derpduck commented:
Fixed the issues mentioned above and updated the original post accordingly.
For the 3 ladders on c2m1 I reverted to the original ladders and fixed their issues instead of replacing them with new ladders.
